extensions/sql-database-projects/src/test/deployDatabaseDialog.test.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as should from 'should';
+import * as path from 'path';
+import * as os from 'os';
+import * as vscode from 'vscode';
+import * as baselines from './baselines/baselines';
+import * as templates from '../templates/templates';
+import { DeployDatabaseDialog } from '../dialogs/deployDatabaseDialog';
+import { Project } from '../models/project';
+import { SqlDatabaseProjectTreeViewProvider } from '../controllers/databaseProjectTreeViewProvider';
+import { ProjectsController } from '../controllers/projectController';
+import { createContext, TestContext } from './testContext';
+
+
+let testContext: TestContext;
+
+describe('Deploy Database Dialog', () => {
+	before(async function (): Promise<void> {
+		testContext = createContext();
+		await templates.loadTemplates(path.join(__dirname, '..', '..', 'resources', 'templates'));
+		await baselines.loadBaselines();
+	});
+
+	it('Should open dialog successfully ', async function (): Promise<void> {
+		const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
+		const projFileDir = path.join(os.tmpdir(), `TestProject_${new Date().getTime()}`);
+
+		const projFilePath = await projController.createNewProject('TestProjectName', vscode.Uri.file(projFileDir), 'BA5EBA11-C0DE-5EA7-ACED-BABB1E70A575');
+		const project = new Project(projFilePath);
+		const deployDatabaseDialog = new DeployDatabaseDialog(project);
+		deployDatabaseDialog.openDialog();
+		should.notEqual(deployDatabaseDialog.deployTab, undefined);
+	});
+
+	it('Should create default database name correctly ', async function (): Promise<void> {
+		const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
+		const projFolder = `TestProject_${new Date().getTime()}`;
+		const projFileDir = path.join(os.tmpdir(), projFolder);
+
+		const projFilePath = await projController.createNewProject('TestProjectName', vscode.Uri.file(projFileDir), 'BA5EBA11-C0DE-5EA7-ACED-BABB1E70A575');
+		const project = new Project(projFilePath);
+
+		const deployDatabaseDialog = new DeployDatabaseDialog(project);
+		should.equal(deployDatabaseDialog.getDefaultDatabaseName(), projFolder);
```

Comment:
default database name (and script name below) should probably be the project name rather than the folder name.  I've been meaning to add that as a property in Project...

Reply:
added it as a property in Project

Reply:
Thanks!